diksha/diksha.controller.js

```diff
@@ -492,80 +493,84 @@ let getHomePage = (req, res) => {
-        }).then(value => {
-            let responses = value.responses;
-            genieResponses = responses['Best of Genie'];
-
-            //	fs.writeFile("/home/admin/Final_thingy", JSON.stringify(genieResponses, null, 4), (err, res) => {});
-
-            return loadSkeletonJson('homePageResponseSkeleton');
-        }).then(value => {
-            responseStructure = value.data;
-            return doThoroughSearch(query);
-        }).then(value => {
-            let responses = value.responses;
-
-            //	console.log("_____HERE____");
-            //	console.log({query});
-            //      console.log({responses});
-
-            return generateResponseStructure(responseStructure, responses);
-        }).then(value => {
-            responseStructure = value.responseStructure;
-            //responseStructure.result.page.sections[i].contents = responses;
-            responseStructure.result.response.sections[0].contents = genieResponses;
-            responseStructure.ts = new Date();
-            responseStructure.ver = parsedReq.ver;
-            responseStructure.id = parsedReq.id;
-            responseStructure.name = parsedReq.request.name;
-            responseStructure.resmsgid = '0211201a-c91e-41d6-ad25-392de813124c';
-
-            //console.log(JSON.stringify(responseStructure, null, 4));
-            //fs.writeFile("/home/admin/api.debug", JSON.stringify(responseStructure), (err, res) => console.log('Written debug info to api.debug'));
+                for (let key in configFilters) {
+                    sectionObject[key] = configFilters[key];  
+                }
+                //console.log({sectionObject});
+	            sectionResponsePromises.push(doSectionwiseSearch(sectionObject));
+            }   
+		return q.all(sectionResponsePromises);				
+	}).then(value => {
+		let responsePromises = [];
+        for(let i in value) {
+            responsePromises.push(getAllPromises(value[i]));
+        }
+        return q.all(responsePromises); 
+	}).then(value => {
+		let responses = {};
+		let defer = q.defer();
+        for(let i in  sectionNames) {
+            responses[i] = value[i].responses.map(response => response.fields)
+        }
+        return responses;	
+	}).then(value => {
+        sectionResponse = value
+		return loadSkeletonJson(reqConfig + 'HomePage')	
+	}).then(value => {
+	    responseStructure = value.data;
+		for(let i in sectionResponse) {	
+            responseStructure.result.response.sections[i].contents.push(...sectionResponse[i]);
+	    }
+		
+		return generateResponseStructure(responseStructure);	
+	}).then(value => {
+        responseStructure = value.responseStructure;
+        responseStructure.ts = new Date();
+        responseStructure.ver = parsedReq.ver;
+        responseStructure.id = parsedReq.id;
+        responseStructure.name = parsedReq.request.name;
+        responseStructure.params.resmsgid = uuidv4();
```

Comment:
Do you not need to specify the length of the id's ?

Reply:
there is no mention of length of id inside the manifest.json

Reply:
But the message id, device id, etc. would have a specific length, wouldn't they? Can you verify this from diksha team and create the string of the appropriate length

Reply:
we will talk to the app team regarding this issue and understand what needs to be done.
Right now there is no msgid or resmsgid coming from their side so we need to ask them about it